### ISSUE TITLE
Problems with React and 'blocks'

### ISSUE THREAD

**Issue by cutterbl** [uNmAnNeR/imaskjs#442]
I'm trying to use 'react-imask' for complex field masking, and running into issues with adding blocks. Consider the following simple setup for a us phone number:

```
import React from 'react';
import IMask from 'imask/esm/imask';
import 'imask/esm/masked/range';
import MaskInput from '@my/mask-input';

import Input from './Input.component';

const blocks = {
  1: {
    mask: IMask.MaskedRange,
    from: 1,
    to: 9,
  },
};

const PhoneInput = ({
  placeholder = '(123) 123-1234',
  mask = '(100) 000-0000',
  ...props
}) => (
  <MaskInput {...{ ...props, placeholder, mask, blocks }} unmask as={Input} />
);
```
The 'MaskInput' is just a wrapper for my third party input control, which I pass in
```
import React from 'react';
import PropTypes from 'prop-types';
import IMask from 'imask/esm/imask';
import { IMaskMixin } from 'react-imask';

const MaskInput = IMaskMixin(({ as: Input, inputRef, ...props }) => (
  <Input inputRef={inputRef} {...props} />
));
export default MaskInput;
```
My issue is the `blocks`. When I'm using a simple `mask` I have no issues. But when I use it with `blocks` my app bombs with a `TypeError` in Storybook, and none of my tests will get past the same errors:
```
TypeError: Cannot call a class as a function
        at _classCallCheck (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/imask/esm/_rollupPluginBabelHelpers-3c58f0e3.js:19:11)
        at MaskedFunction.MaskedRange (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/imask/esm/masked/range.js:22:5)
        at MaskedFunction.doValidate (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/masked/base.js:266:38)
        at MaskedFunction._appendChar (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/masked/base.js:147:29)
        at MaskedPattern._appendCharRaw (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/masked/pattern.js:220:35)
        at MaskedPattern._appendChar (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/masked/base.js:143:26)
        at MaskedPattern.append (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/masked/base.js:193:32)
        at MaskedPattern.resolve (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/masked/base.js:78:12)
        at MaskedPattern.set (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/masked/base.js:324:12)
        at set (<anonymous>)
        at set (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/_rollupPluginBabelHelpers-3c58f0e3.js:204:10)
        at _set (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/_rollupPluginBabelHelpers-3c58f0e3.js:208:11)
        at MaskedPattern.set (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/masked/pattern.js:686:7)
        at InputMask.updateValue (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/controls/input.js:145:7)
        at new InputMask (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/controls/input.js:59:10)
        at IMask (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/imask/esm/core/holder.js:11:10)
        at MaskedComponent.initMask (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/react-imask/esm/mixin-c3942261.js:228:24)
        at MaskedComponent.componentDidMount (/Users/Foo/Documents/MyApp/packages/@my/mask-input/node_modules/react-imask/esm/mixin-c3942261.js:178:14)
        at commitLifeCycles (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:20663:24)
        at commitLayoutEffects (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:23426:7)
        at HTMLUnknownElement.callCallback (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:3945:14)
        at HTMLUnknownElement.callTheUserObjectsOperation (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/jsdom/lib/jsdom/living/generated/EventListener.js:26:30)
        at innerInvokeEventListeners (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:318:25)
        at invokeEventListeners (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:274:3)
        at HTMLUnknownElementImpl._dispatch (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:221:9)
        at HTMLUnknownElementImpl.dispatchEvent (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:94:17)
        at HTMLUnknownElement.dispatchEvent (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:231:34)
        at Object.invokeGuardedCallbackDev (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:3994:16)
        at invokeGuardedCallback (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:4056:31)
        at commitRootImpl (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:23151:9)
        at unstable_runWithPriority (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/scheduler/cjs/scheduler.development.js:646:12)
        at runWithPriority$1 (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:11276:10)
        at commitRoot (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:22990:3)
        at performSyncWorkOnRoot (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:22329:3)
        at scheduleUpdateOnFiber (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:21881:7)
        at updateContainer (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:25482:3)
        at /Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:26021:7
        at unbatchedUpdates (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:22431:12)
        at legacyRenderSubtreeIntoContainer (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:26020:5)
        at Object.render (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:26103:10)
        at /Users/Foo/Documents/MyApp/packages/my-components/node_modules/@testing-library/react/dist/pure.js:99:25
        at batchedUpdates$1 (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom.development.js:22380:12)
        at act (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/react-dom/cjs/react-dom-test-utils.development.js:1042:14)
        at render (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/@testing-library/react/dist/pure.js:95:26)
        at renderPhone (/Users/Foo/Documents/MyApp/packages/my-components/src/form/controls/input/PhoneInput.spec.js:20:31)
        at Object.<anonymous> (/Users/Foo/Documents/MyApp/packages/my-components/src/form/controls/input/PhoneInput.spec.js:30:29)
        at Object.asyncJestTest (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:106:37)
        at /Users/Foo/Documents/MyApp/packages/my-components/node_modules/jest-jasmine2/build/queueRunner.js:45:12
        at new Promise (<anonymous>)
        at mapper (/Users/Foo/Documents/MyApp/packages/my-components/node_modules/jest-jasmine2/build/queueRunner.js:28:19)
        at /Users/Foo/Documents/MyApp/packages/my-components/node_modules/jest-jasmine2/build/queueRunner.js:75:41
```
I'm following the documentation as much as I can, but feel like I'm missing something critical here. Is it some import? What am I doing wrong?

**Comment by cutterbl** [comment]
Additional Info: I created a [CodeSandbox](https://codesandbox.io/s/react-imask-change-test-uyb8u?file=/src/PhoneInput.component.js), using the above basic design, and it worked there fine. This makes me think that maybe theirs something wrong/missing with my Babel config. I use a very basic config locally:
```
// babel.config.js
module.exports = function (api) {
  api.cache(false);
  return {
    presets: ['@babel/preset-env', 'react-app'],
    plugins: ['@babel/plugin-transform-runtime'],
    env: {
      test: {
        presets: [
          [
            '@babel/preset-env',
            {
              modules: 'commonjs',
              targets: {
                node: 'current',
              },
            },
          ],
          'react-app',
        ],
      },
    },
  };
};
```
Again, any help here is greatly appreciated.

**Comment by cutterbl** [comment]
Could it be possible that it's because I am not having babel compile external dependencies? If I look at `range.js` in DevTools, from Storybook I see it exactly as it is in `node_modules`, but when I look at it from CodeSandbox I see it as it if it's been transpiled?

**Comment by uNmAnNeR** [comment]
@cutterbl
1) try not using `esm` stuff, just import regular components from `imask` and `react-imask`.
2) v6.0.6 might solve your problem (will release soon)

**Comment by cutterbl** [comment]
I tried importing the regular imask. Didn't resolve the issue. If there's a branch/package for 6.0.6 I could try that...

**Comment by cutterbl** [comment]
OK, I tried 6.0.6. First thing I notice is that there's no longer a `imask/esm/masked` path. The `package.json`'s `module` path points to `esm/index.js`, which doesn't exist in the package. (For reference, I'm not using TypeScript either). Doing `import IMask from 'imask';` still give me the same errors when I try to use a block, when I start typing in a field.

**Comment by uNmAnNeR** [comment]
@cutterbl reverted back `esm` stuff in 6.0.7, was a problem with some rollup plugin after updating dependencies. please try again

**Comment by cutterbl** [comment]
So, I was able to get 6.0.7 to work within Storybook, but still receive the same errors in Jest when testing. I was able to use the `import IMask from 'imask'`, but not `import IMask from 'imask/esm/imask'`

Until I setup a `moduleNameMapper`, pointing `imask/esm/imask` to `node_modules/imask/dist/imask.js`. This took care of the standard import. This meant that I had to use `import IMask from 'imask/esm/imask'` everywhere, and I could not do things like `import MaskedRange from `imask/esm/masked/range`', as I can't map those and have them properly load.

But, it's a start.
